scripts/mutect2_wdl/unsupported/mutect2_opt.wdl

```diff
@@ -214,117 +193,93 @@ task M2 {
-  String m2_docker
   File? gatk4_jar_override
-  Int preemptible_attempts
   String? m2_extra_args
   Boolean? is_bamOut
-  
-  # Do not populate this unless you know what you are doing...
   File? auth
 
+  # Runtime parameters
+  Int? mem
+  String gatk_docker
+  Int? preemptible_attempts
+  Int? disk_space_gb
+
   command <<<
 
-  if [[ "${auth}" == *.json ]]; then
-    gsutil cp ${auth} /root/.config/gcloud/application_default_credentials.json
-    GOOGLE_APPLICATION_CREDENTIALS=/root/.config/gcloud/application_default_credentials.json
-    export GOOGLE_APPLICATION_CREDENTIALS=/root/.config/gcloud/application_default_credentials.json
-  fi
+    if [[ "${auth}" == *.json ]]; then
+      gsutil cp ${auth} /root/.config/gcloud/application_default_credentials.json
+      GOOGLE_APPLICATION_CREDENTIALS=/root/.config/gcloud/application_default_credentials.json
+      export GOOGLE_APPLICATION_CREDENTIALS=/root/.config/gcloud/application_default_credentials.json
+    fi
 
   # Use GATK Jar override if specified
   GATK_JAR=${gatk4_jar}
   if [[ "${gatk4_jar_override}" == *.jar ]]; then
       GATK_JAR=${gatk4_jar_override}
   fi
 
+  java -Xmx4g -jar $GATK_JAR GetSampleName -I ${tumor_bam} -O tumor_name.txt
+  output_vcf=`cat tumor_name.txt`-tumor-only.vcf
+
+   # We need to create a file regardless, even if it stays empty
+   touch bamout.bam
+
   if [[ "_${normal_bam}" == *.bam ]]; then
       java -Xmx4g -jar $GATK_JAR GetSampleName -I ${normal_bam} -O normal_name.txt
       normal_command_line="-I ${normal_bam} -normal `cat normal_name.txt`"
-  else
-      # Note that normal_name.txt is always created, it's just empty if no normal sample was given.
-      #     This is done to allow an output parameter of the normal sample.
-      touch normal_name.txt
+      output_vcf=`cat tumor_name.txt`-vs-`cat normal_name.txt`.vcf
   fi
 
-  java -Xmx4g -jar $GATK_JAR GetSampleName -I ${tumor_bam} -O tumor_name.txt
 
-  # We need to create a file regardless, even if it stays empty
-  touch bamout.bam
 
   java -Xmx4g -jar $GATK_JAR Mutect2 \
     -R ${ref_fasta} \
     -I ${tumor_bam} \
     -tumor `cat tumor_name.txt` \
     $normal_command_line \
-    ${"--germline_resource " + gnomad} \
-    ${"--normal_panel " + pon} \
+    ${"--germline-resource " + gnomad} \
+    ${"-pon " + pon} \
     ${"-L " + intervals} \
-    -O "${output_vcf_name}.vcf" \
+    -O $output_vcf \
     ${true='--bamOutput bamout.bam' false='' is_bamOut} \
     ${m2_extra_args}
   >>>
 
   runtime {
-      docker: "${m2_docker}"
-      memory: "5 GB"
-      disks: "local-disk " + 200 + " HDD"
-      preemptible: "${preemptible_attempts}"
+        docker: "${gatk_docker}"
+        memory: select_first([mem, 5]) + " GB"
+        disks: "local-disk " + select_first([disk_space_gb, 100]) + " HDD"
+        preemptible: select_first([preemptible_attempts, 2])
   }
 
   output {
-    File output_vcf = "${output_vcf_name}.vcf"
+    File output_vcf = glob("*.vcf")[0]
```

Comment:
Hey, @jsotobroad and @meganshand , how do you feel about this glob statement?

Reply:
We don't really like the globbed output for just one file but we can see why you would want to do it.  We had another way to get rid of all the workflow input sample names and no more preprocessing step.  We were thinking of taking out the `GetSampleName` calls out of M2 into its own task (because its awkward calling the same exact thing for every scatter of M2 and they all return the same output).  Use the output of this new GetSampleName task to get the vcf output name with something like

```
  String output_vcf = if defined(normal_bam) then GetSampleNameTask.normal_sample_name + "-vs-" + GetSampleNameTask.tumor_sample_name else GetSampleNameTask.tumor_sample_name + "-tumor-only"
```

and use this string as as the output vcf name as it was used in the original wdl.  Does that sound like a good solution to you guys?

Reply:
@LeeTL1220 @jsotobroad I was trying to completely eliminate getting the sample name, thereby getting rid of one task entirely.  I've heard that there is overhead for every task on the cloud and I was perhaps being paranoid.  If this isn't a big deal I will do it Jose's way.

Reply:
Money wise it is very inexpensive deal to run and workflow wise, because the workflow has to run `SplitIntervals` before it can run M2 the total runtime of the workflow shouldn't be affected either.  It's definitely something that should be able to made nice with future cromwell/wdl features hopefully....

Reply:
I'm satisfied and will do it @jsotobroad's recommended way.